### PR TITLE
Add QCArchive datastructures

### PR DESCRIPTION
**Pull request by wiederm**
## Description
We want to provide a dataset structure for the most common datasets that are provided through the QCArchive infrastructure. These datasets should play nicely with the `pytorch` `dataloader` objects and inherit from `torch.utils.data.Dataset`.

## Todos
  - [x] Outline general structure 
  - [x] Implement QM9 toy dataset
  - [ ] Tests are passing
 
## Questions
- [ ] How to best query different labels?
- [ ] Is there any use case for different datapoints beside coordinates?
- [ ] Is a `view` the best structure to perform queries on? 

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by chrisiacovella** [comment]
I think, after a few minor changes, we should just merge this PR so we can create a separate one aimed at fixing the CI workflow.

**Review by chrisiacovella — CHANGES_REQUESTED** [review]
I think this is ready after:
1- remove qm9.py
2- remove the "to_file" call (as that is going to break things
3- see if the quick change to the test environment file fixes the failing CI

File: devtools/conda-envs/test_env.yaml [hunk 1]
```diff
@@ -8,13 +8,19 @@ dependencies:
     # Base depends
   - python
   - pip
+  - h5py
+  - requests
+  - tqdm
+  - qcarchive
```
[the remark this answers is not in the document]
Comment (wiederm) [reply devtools/conda-envs/test_env.yaml:14]: agreed, will change!

[the remark this answers is not in the document]
Comment (wiederm) [reply devtools/conda-envs/test_env.yaml:14]: updated import

File: modelforge/dataset/dataset.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    """
+
+    def __init__(self, dataset_file: str = None, name: str = ""):
+        self.dataset_file = dataset_file
+        self.name = name
+        self.dataset = self.load(dataset_file)
+        (
+            self.molecules,
+            self.records,
+        ) = self.dataset.get_molecules(), self.dataset.get_records(method="b3lyp")
```
[the remark this answers is not in the document]
Comment (wiederm) [reply modelforge/dataset/dataset.py:21]: agreed! modifyied the method accordinly.

File: modelforge/dataset/dataset.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        except Exception:
+            print(
+                f"Dataset {qcportal_data['dataset']} is not available in collection {qcportal_data['collection']}."
+            )
+
+        if dataset_file and os.path.isfile(dataset_file):
+            if not dataset_file.endswith(".hdf5"):
+                raise ValueError("Input file must be an .hdf5 file.")
+            logger.debug(f"Loading from {dataset_file}")
+            dataset.set_view(dataset_file)  # NOTE: this doesn't work yet
```
[the remark this answers is not in the document]
Comment (wiederm) [reply modelforge/dataset/dataset.py:55]: oh, that is resolved now.

File: modelforge/dataset/qm9.py
[the remark this answers is not in the document]
Comment (wiederm) [reply modelforge/dataset/qm9.py:1]: thanks!

**Review by chrisiacovella — APPROVED** [review]
Looks good. Once merged, I'll get a PR going trying to resolve the CI issues.
